R/parse_prepare.R

```diff
@@ -0,0 +1,115 @@
+parse_prepare <- function(quo, input, call) {
```

Comment:
Docs will help in due course - but is there a better name for `quo` ?

EDIT: Oh, it's a quosure. :face_palm: 

Is input really input_type?

Reply:
This one thankfully never gets exported to user; the args for `odin_parse` are close to what the user will use.  Happy to go with `input_type` if you think that's clearer